[RDY] tests: Migrate legacy test 57.

lucc opened the pull request.

lucc committed:
tests: Migrate legacy test 57.

fwalch commented on test/functional/legacy/057_spec.lua:
```diff
@@ -0,0 +1,973 @@
+      t28: done
+      ]=])
+
+    --execute('so small.vim')
```
You can just remote this line entirely, no need to have this as a comment.

fwalch commented on test/functional/legacy/057_spec.lua:
```diff
@@ -0,0 +1,973 @@
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('57', function()
```
Change to `describe('sort command')` (and rename file to `057_sort_spec.lua`)?

fwalch commented on test/functional/legacy/057_spec.lua:
```diff
@@ -0,0 +1,973 @@
+
+    --execute('so small.vim')
+
+    source([[
```
I think it would be better to use `execute()` here (AFAIR you can execute multiple commands after one another if you chain them with commas, like `execute('/.../', '/...')`. `source` will write a temporary file, which is a bit of overhead I think we can avoid here.

lucc commented:
Thanks @fwalch I removed the unnecessary comment, and changed the description of the test and the file name. I am still working on the usage of `execute()` with several arguments (from the source in `helpers.lua` I take that it should work, but the test fails if I try it.).

fwalch commented:
> I am still working on the usage of execute() with several arguments (from the source in helpers.lua I take that it should work, but the test fails if I try it.

It's okay if you just use `source()`, sorry for bringing that up and causing confusion.

lucc commented:
rebased

lucc commented:
rebased and minor improvments

lucc commented:
Rebased to restart travis build.

Should I reference #1286 in the legacy test migrating PRs?

justinmk commented:
Sure.

Pyrohh commented:
@lucc could you rebase to fix the merge conflict? Besides that, I think these should be split into `it()` blocks, given they're already organized (and labelled) as pseudo-separated tests.

lucc committed:
tests: Split migrated test 57.

lucc commented:
@Pyrohh done

Pyrohh commented:
LGTM, I'll merge soon if there's no more comments.

Pyrohh commented:
These improvements are great @lucc, I'll wait for the tests to complete and merge.

Pyrohh commented:
Thanks @lucc.
